trailblaze-report/src/main/java/xyz/block/trailblaze/report/GenerateTestResultsCliCommand.kt

```diff
@@ -0,0 +1,406 @@
+    if (verbose) {
+      printDetailedResults(sessionResults)
+      printFailedTests(sessionResults)
+    }
+
+    val summaryReport = CiSummaryReport(
+      metadata = metadata,
+      results = sessionResults,
+    )
+
+    val output = outputArg ?: File(logsDir, "trailblaze_test_report.${outputFormat.extension}")
+    val content = when (outputFormat) {
+      OutputFormat.JSON -> json.encodeToString(value = summaryReport)
+      OutputFormat.YAML -> yamlSerializer.encodeToString(value = summaryReport)
+    }
+
+    output.writeText(content)
+    println()
+    println("📄 Summary written to: ${output.absolutePath}")
+  }
+
+  private fun printFailedTests(results: List<SessionResult>) {
+    val failedTests = results.filter { it.outcome == Outcome.FAILED }
+
+    val output = buildString {
+      if (failedTests.isEmpty()) {
+        appendLine()
+        appendLine("✅ No failed tests!")
+        appendLine()
+      } else {
+        appendLine()
+        appendLine("❌ FAILED TESTS (${failedTests.size})")
+        appendLine("────────────────────────────────────────────────────────────")
+
+        failedTests.forEach { result ->
+          val platformIcon = if (result.platform.contains("android", ignoreCase = true)) "🤖" else "🍎"
+
+          appendLine()
+          appendLine("❌ ${result.title}")
+          appendLine("   Platform: $platformIcon ${result.platform}")
+          appendLine("   Duration: ${formatDuration(result.duration_ms)}")
+          if (result.failure_reason != null) {
+            appendLine("   Reason: ${result.failure_reason}")
+          }
+          appendLine("   Session: ${result.session_id}")
+        }
+        appendLine()
+      }
+    }
+    print(output)
+  }
+
+  private fun buildMetadataFromEnvironment(): CiRunMetadata {
+    fun getEnv(name: String): String? = System.getenv(name)?.takeIf { it.isNotBlank() }
+    fun getEnvList(name: String): List<String> = getEnv(name)?.split(",")?.map {
+        it.trim()
+      }?.filter {
+        it.isNotEmpty()
+      } ?: emptyList()
+
+    return CiRunMetadata(
+      target_app = getEnv("TRAILBLAZE_TARGET_APP") ?: "",
+      build_type = getEnv("TRAILBLAZE_BUILD_TYPE") ?: "",
+      devices = getEnvList("TRAILBLAZE_DEVICES"),
+      android_build_url = getEnv("ANDROID_BUILD_URL"),
+      ios_build_url = getEnv("IOS_BUILD_URL"),
+      retry_count = getEnv("TRAILBLAZE_TEST_RETRY_COUNT")?.toIntOrNull() ?: 0,
+      ai_enabled = getEnv("TRAILBLAZE_AI_ENABLED")?.toBoolean() ?: true,
+      ai_fallback_enabled = getEnv("TRAILBLAZE_AI_FALLBACK_ENABLED")?.toBoolean() ?: true,
+      parallel_execution = getEnv("TRAILBLAZE_PARALLEL_EXECUTION")?.toBoolean() ?: false,
+      ci_build_url = getEnv("BUILDKITE_BUILD_URL") ?: getEnv("CI_BUILD_URL"),
+      ci_build_number = getEnv("BUILDKITE_BUILD_NUMBER") ?: getEnv("CI_BUILD_NUMBER"),
+      git_commit = getEnv("BUILDKITE_COMMIT") ?: getEnv("GIT_COMMIT"),
+      git_branch = getEnv("BUILDKITE_BRANCH") ?: getEnv("GIT_BRANCH"),
```

Comment:
Don't love that it has `BUILDKITE_`, but does have the standard `GIT_` and `CI_` fallback, so it's okay.  Ideally if someone has their own build system, they could just copy over the environment vars, but this won't break anyone, and might help someone on buildkite 👍